applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.h

```diff
@@ -12,42 +12,40 @@
 
 #pragma once
 
-// Project includes
-#include "custom_constitutive/linear_elastic_plane_strain_K0_law.h"
+#include "custom_constitutive/linear_elastic_law.h"
+#include "geo_mechanics_application_constants.h"
 
 namespace Kratos
 {
 
+class ConstitutiveType;
+
 /**
  * @class GeoLinearElasticPlaneStrain2DLaw
  * @ingroup GeoMechanicsApplication
  * @brief This class defines a small deformation linear elastic constitutive model for plane strain cases
- * @details This class derives from the linear elastic case on 3D
  * @author Vahid Galavi
  */
-class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoLinearElasticPlaneStrain2DLaw : public LinearPlaneStrainK0Law
+class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoLinearElasticPlaneStrain2DLaw : public GeoLinearElasticLaw
 {
 public:
-    /// The base class LinearPlaneStrainK0Law type definition
-    using BaseType = LinearPlaneStrainK0Law;
-
-    /// The size type definition
-    using SizeType = std::size_t;
-
-    /// Static definition of the dimension
+    using BaseType                      = GeoLinearElasticLaw;
+    using SizeType                      = std::size_t;
     static constexpr SizeType Dimension = N_DIM_2D;
-
-    /// Static definition of the VoigtSize
-    // for the time being
     static constexpr SizeType VoigtSize = VOIGT_SIZE_2D_PLANE_STRAIN;
 
-    /// Counted pointer of LinearPlaneStrainK0Law
     KRATOS_CLASS_POINTER_DEFINITION(GeoLinearElasticPlaneStrain2DLaw);
+    GeoLinearElasticPlaneStrain2DLaw();
 
-    /**
-     * @brief The clone operation
-     */
-    ConstitutiveLaw::Pointer Clone() const override;
+    explicit GeoLinearElasticPlaneStrain2DLaw(std::unique_ptr<ConstitutiveType> pConstitutiveDimension);
+    GeoLinearElasticPlaneStrain2DLaw(const GeoLinearElasticPlaneStrain2DLaw& rOther);
+    GeoLinearElasticPlaneStrain2DLaw& operator=(GeoLinearElasticPlaneStrain2DLaw const& rOther);
```

Comment:
We always put `const` in front of the type name:
```suggestion
    GeoLinearElasticPlaneStrain2DLaw& operator=(const GeoLinearElasticPlaneStrain2DLaw& rOther);
```

Reply:
Done